src/test/java/cmput402/tdd/AppTest.java

```diff
@@ -265,4 +265,269 @@ public void testDisplayPeople() {
         assertEquals(app.displayPeople(), out);
         
     }
+  
+    @Test
+    public void testEditRecipe(){
+        App app = new App();
+        Recipe recipe = new Recipe("apple pie");
+
+        Item item1 = new Item("apple", 4.0f);
```

Comment:
```suggestion
        Item apple = mock(Item.class);
        when(apple.getName()).thenReturn("apple");
        when(apple.getCost()).thenReturn(4.0f);
```

Reply:
Thanks for suggestion, made changes

Reply:
After testing out the mocking of the items and recipes, I realized it would not be possible to mock the item objects. This is because in our App, we call `createItem` which makes a REAL item object, when we use `recipe.items.get(item1)` where item1 is a MOCKED object then it will return null. This is because using `.get()` we are unable to find key of the mocked object since only the real item1. Maybe someone can clarify how I would implement this, for now i'll leave it as is.

Reply:
You are mostly correct you cannot use mock objects since the code itself creates real objects. It is possible to use it for some cases where we do not remove